### PR TITLE
add nsync + test

### PR DESCRIPTION
**Pull request by headupinclouds**
Note:
* TODO: replace include_directories() -> target_include_directories(), etc
* OSX build has numerous deprecated function warnings
```
note: 'OSAtomicCompareAndSwap32Barrier' has been explicitly marked deprecated here
bool    OSAtomicCompareAndSwap32Barrier( int32_t __oldValue, int32_t __newValue, volatile int32_t *__theValue );
```

https://github.com/ruslo/hunter/issues/1155

- [x] has hunter-packages/nsync

### CODE REVIEW

File: cmake/projects/nsync/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    nsync
+    VERSION
+    1.14-p0
+    URL
+    "https://github.com/headupinclouds/nsync/archive/v1.14-p0.tar.gz"
```
Comment (ruslo) [inline cmake/projects/nsync/hunter.cmake]: Fork: https://github.com/hunter-packages/nsync

Comment (headupinclouds) [reply cmake/projects/nsync/hunter.cmake]: Moved

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2727
* https://travis-ci.org/ingenue/hunter/builds/300064610?utm_source=github_status&utm_medium=notification

File: cmake/projects/nsync/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    nsync
+    VERSION
+    1.14-p0
+    URL
+    "https://github.com/hunter-packages/nsync/archive/v1.14-p0.tar.gz"
```
Comment (ruslo) [inline cmake/projects/nsync/hunter.cmake]: Use this release: https://github.com/hunter-packages/nsync/releases/tag/v1.14-p1

Comment (headupinclouds) [reply cmake/projects/nsync/hunter.cmake]: Ouch.  Thanks.  Done. Test == `SUCCESS`

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2737
* https://travis-ci.org/ingenue/hunter/builds/300150305?utm_source=github_status&utm_medium=notification

**Comment by headupinclouds** [comment]
* vs-14-2015 is working, all others fail 
* NDK toolchains are failing due to:
```
/home/travis/build/ingenue/hunter/_testing/Hunter/_Base/9e18f06/97b2613/b5d6ba7/Build/nsync/Source/testing/array.c:15:10: fatal error: 'platform.h' file not found
#include "platform.h"
         ^~~~~~~~~~~~
```

**Comment by ruslo** [comment]
I will exclude everything that is not working for now. Later we can apply fixes.

**Comment by headupinclouds** [comment]
Ok.  I've added some initial Android patches in a PR for discussion here https://github.com/hunter-packages/nsync/pull/1

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2746
* https://travis-ci.org/ingenue/hunter/builds/300323041?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Follow-up: https://github.com/ruslo/hunter/commit/f16afee092690b94c6e9184e1b913692827968b2

Released: https://github.com/ruslo/hunter/releases/tag/v0.19.148

**Pull request merged**